modules/webbrowser/src/processors/progressbarobservercef.cpp

```diff
@@ -0,0 +1,63 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2019 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <modules/webbrowser/processors/progressbarobservercef.h>
+
+#include <inviwo/core/processors/progressbarowner.h>
+
+namespace inviwo {
+
+ProgressBarObserverCEF::ProgressBarObserverCEF(CefRefPtr<CefFrame> frame,
+                                               std::string onProgressChange,
+                                               std::string onVisibleChange)
+    : onProgressChange_(onProgressChange), onVisibleChange_(onVisibleChange), frame_(frame) {}
+
+void ProgressBarObserverCEF::progressChanged(float progress) {
+    // Frame might be null if for example webpage is not found on startup
+    if (!frame_ || getOnProgressChange().empty()) {
+        return;
+    }
+    std::stringstream script;
+    script << getOnProgressChange() << "(" << progress << ");";
+    frame_->ExecuteJavaScript(script.str(), frame_->GetURL(), 0);
+}
+
+void ProgressBarObserverCEF::progressBarVisibilityChanged(bool visible) {
+    // Frame might be null if for example webpage is not found on startup
+    if (!frame_ || getOnVisibleChange().empty()) {
```

Comment:
`getOnVisibleChange` ? No hint that this is connected to the progress bar...

Reply:
Renamed